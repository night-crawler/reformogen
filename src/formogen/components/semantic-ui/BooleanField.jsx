import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { Form, Checkbox } from 'semantic-ui-react';

import { errorsType, layoutOptsType } from '../fieldPropTypes';
import Label from './Label';

import { MessageList } from './MiscComponents';


BooleanField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.bool,
    help_text: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    widget: PropTypes.string,
    errors: errorsType,

    required: PropTypes.bool,
    editable: PropTypes.bool,

    helpTextOnHover: PropTypes.bool,
    layoutOpts: layoutOptsType,

    updateProps: PropTypes.func,
    onChange: PropTypes.func,
};
export default function BooleanField(props) {
    const handleChange = (e, newValue) => {
        props.onChange(e, { name: props.name, value: newValue.checked });
    };

    const { widget = '' } = props;

    let _props = {
        name: props.name,
        checked: props.value,
        placeholder: props.placeholder,
        onChange: handleChange,
        toggle: widget.toLowerCase() === 'toggle' || undefined,
        slider: widget.toLowerCase() === 'slider' || undefined,
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <Checkbox { ..._props } />
            { !props.helpTextOnHover ? <span className='help-text'>{ props.help_text }</span> : '' }
            <MessageList messages={ props.errors } />
        </Form.Field>
    );
}
