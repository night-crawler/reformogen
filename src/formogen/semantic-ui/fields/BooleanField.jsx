import React from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import { Form, Checkbox } from 'semantic-ui-react';

import { errorsType, layoutOptsType } from '../../fieldPropTypes';
import { FieldLabel } from '../FieldLabel';
import { ErrorsList } from '../ErrorsList';


BooleanField.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  help_text: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  widget: PropTypes.string,
  errors: errorsType,

  required: PropTypes.bool,
  editable: PropTypes.bool,

  helpTextOnHover: PropTypes.bool,
  layoutOpts: layoutOptsType,

  onChange: PropTypes.func,
};
BooleanField.defaultProps = {
  widget: '',
};
export function BooleanField(props) {
  const handleChange = (e, newValue) => {
    props.onChange(e, { name: props.name, value: newValue.checked });
  };

  return (
    <Form.Field
      required={ props.required }
      disabled={ !props.editable }
      width={ props.layoutOpts.width }
      error={ !isEmpty(props.errors) }
    >
      <FieldLabel { ...props } />
      <Checkbox 
        name={ props.name }
        checked={ !!props.value || false }
        placeholder={ props.placeholder }
        onChange={ handleChange }
        toggle={ props.widget.toLowerCase() === 'toggle' || undefined }
        slider={ props.widget.toLowerCase() === 'slider' || undefined }
      />
      { !props.helpTextOnHover
        ? <span className='help-text'>{ props.help_text }</span>
        : ''
      }
      <ErrorsList messages={ props.errors } />
    </Form.Field>
  );
}
