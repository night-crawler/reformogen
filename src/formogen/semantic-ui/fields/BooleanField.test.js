import { BooleanField } from './BooleanField';


it('<BooleanField />', () => {
  const wrapper = shallow(
    <BooleanField
      type='BooleanField'
      name='BooleanField: name'
      verbose_name='BooleanField: verbose_name'
      help_text='BooleanField: help_text'
      layoutOpts={ { width: 1 } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
