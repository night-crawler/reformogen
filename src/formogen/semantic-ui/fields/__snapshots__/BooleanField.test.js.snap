// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooleanField /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BooleanField
    help_text="BooleanField: help_text"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="BooleanField: name"
    type="BooleanField"
    verbose_name="BooleanField: verbose_name"
    widget=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FieldLabel
          help_text="BooleanField: help_text"
          layoutOpts={
            Object {
              "width": 1,
            }
          }
          name="BooleanField: name"
          type="BooleanField"
          verbose_name="BooleanField: verbose_name"
          widget=""
        />,
        <Checkbox
          checked={false}
          name="BooleanField: name"
          onChange={[Function]}
          type="checkbox"
        />,
        <span
          className="help-text"
        >
          BooleanField: help_text
        </span>,
        <ErrorsList
          color="red"
          header="Errors"
        />,
      ],
      "disabled": true,
      "error": false,
      "required": undefined,
      "width": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "help_text": "BooleanField: help_text",
          "layoutOpts": Object {
            "width": 1,
          },
          "name": "BooleanField: name",
          "type": "BooleanField",
          "verbose_name": "BooleanField: verbose_name",
          "widget": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": false,
          "name": "BooleanField: name",
          "onChange": [Function],
          "placeholder": undefined,
          "slider": undefined,
          "toggle": undefined,
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BooleanField: help_text",
          "className": "help-text",
        },
        "ref": null,
        "rendered": "BooleanField: help_text",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
          "header": "Errors",
          "messages": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FieldLabel
            help_text="BooleanField: help_text"
            layoutOpts={
              Object {
                "width": 1,
              }
            }
            name="BooleanField: name"
            type="BooleanField"
            verbose_name="BooleanField: verbose_name"
            widget=""
          />,
          <Checkbox
            checked={false}
            name="BooleanField: name"
            onChange={[Function]}
            type="checkbox"
          />,
          <span
            className="help-text"
          >
            BooleanField: help_text
          </span>,
          <ErrorsList
            color="red"
            header="Errors"
          />,
        ],
        "disabled": true,
        "error": false,
        "required": undefined,
        "width": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "help_text": "BooleanField: help_text",
            "layoutOpts": Object {
              "width": 1,
            },
            "name": "BooleanField: name",
            "type": "BooleanField",
            "verbose_name": "BooleanField: verbose_name",
            "widget": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "name": "BooleanField: name",
            "onChange": [Function],
            "placeholder": undefined,
            "slider": undefined,
            "toggle": undefined,
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BooleanField: help_text",
            "className": "help-text",
          },
          "ref": null,
          "rendered": "BooleanField: help_text",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "red",
            "header": "Errors",
            "messages": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
